src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -296,10 +296,13 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
           this._setProgressiveSource();
         }
         if (this._sourceObj && this._sourceObj.url) {
-          this._setSrc();
+          this._setSrc().then(() => {
+            this._videoElement.load();
+          });
           this._trigger(CustomEventType.ABR_MODE_CHANGED, {mode: this._isProgressivePlayback() ? 'manual' : 'auto'});
```

Comment:
to keep it consistent we need to also move the event to the promise.
Please check that code flow and event order hasn't changed as well - both internal handling and analytics.
Also, as `_setSrc`is called from the select progressive source, then please check that this is also working.
@yairans please review.

Reply:
What do you mean by checking the event order? compare to previous version?

Reply:
Yes, please check that this async flow didn't change anything - low reason to think it did, but need to verify this.

Reply:
I verified the events, i didn't find different